lib/experimental/blocks.php

```diff
@@ -77,3 +77,26 @@ function wp_enqueue_block_view_script( $block_name, $args ) {
 		add_filter( 'render_block', $callback, 10, 2 );
 	}
 }
+
+
+/**
+ * Registers the metadata block attribute for block types.
+ *
+ * @param array $args Array of arguments for registering a block type.
+ * @return array $args
+ */
+function gutenberg_register_metadata_attribute( $args ) {
+	// Setup attributes if needed.
+	if ( ! isset( $args['attributes'] ) || ! is_array( $args['attributes'] ) ) {
```

Comment:
Independent of this PR that moves the existing functionality in `experimental` folder, I think we need to follow the `supports` approach, that means checking `block_has_support` and not just existing attribute.

Was that discussed in the original PR and we want every block to have this support?

Reply:
I don't think it was discussed but it seems reasonable to use `block_has_support` on this as well.

Reply:
@ntsekouras Is the block supports required to merge this PR?

Reply:
No, it's out of this PR's scope.